domain/secretbackend/service/service.go

```diff
@@ -0,0 +1,580 @@
+			ControllerUUID: s.controllerUUID,
+			ModelUUID:      m.ID.String(),
+			ModelName:      m.Name,
+			BackendConfig:  *k8sConfig,
+		}
+		if currentBackend.Name == provider.Auto {
+			info.ActiveID = k8sBackendID
+		}
+	}
+
+	backends, err := s.st.ListSecretBackends(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, b := range backends {
+		if b.Name == currentBackend.Name {
+			info.ActiveID = b.ID
+		}
+		info.Configs[b.ID] = provider.ModelBackendConfig{
+			ControllerUUID: s.controllerUUID,
+			ModelUUID:      m.ID.String(),
+			ModelName:      m.Name,
+			BackendConfig: provider.BackendConfig{
+				BackendType: b.BackendType,
+				Config:      convertConfigToAny(b.Config),
+			},
+		}
+	}
+	if info.ActiveID == "" {
+		return nil, fmt.Errorf("%w: %q", secretbackenderrors.NotFound, currentBackend.Name)
+	}
+	return &info, nil
+}
+
+func convertConfigToAny(config map[string]string) map[string]interface{} {
+	if len(config) == 0 {
+		return nil
+	}
+	result := make(map[string]interface{}, len(config))
+	for k, v := range config {
+		result[k] = v
+	}
+	return result
+}
+
+func convertConfigToString(config map[string]interface{}) map[string]string {
+	if len(config) == 0 {
+		return nil
+	}
+	result := make(map[string]string, len(config))
+	for k, v := range config {
+		if s, ok := v.(string); ok {
+			result[k] = s
+			continue
+		}
+		result[k] = fmt.Sprintf("%v", v)
+	}
+	return result
+}
+
+// GetSecretBackendConfigLegacy gets the config needed to create a client to secret backends.
+// TODO - drop when we no longer support juju 3.1.x
+func (s *Service) GetSecretBackendConfigLegacy(
+	ctx context.Context, modelUUID coremodel.UUID, cloud cloud.Cloud, cred cloud.Credential,
+) (*provider.ModelBackendConfigInfo, error) {
+	// TODO: implement once we have secret service in place.
+	return nil, nil
+}
+
+// GetSecretBackendConfig returns the secret backend configuration for the given backend ID.
+func (s *Service) GetSecretBackendConfig(
+	ctx context.Context, modelUUID coremodel.UUID, cloud cloud.Cloud, cred cloud.Credential,
+) (*provider.ModelBackendConfigInfo, error) {
+	// TODO: implement once we have secret service in place.
+	return nil, nil
+}
+
+// GetSecretBackendConfigForDrain returns the secret backend configuration
+// for the given backend ID for the drain worker.
+func (s *Service) GetSecretBackendConfigForDrain(
+	ctx context.Context, backendID string,
+	modelUUID coremodel.UUID, cloud cloud.Cloud, cred cloud.Credential,
+) (*provider.ModelBackendConfigInfo, error) {
+	// TODO: implement once we have secret service in place.
+	return nil, nil
+}
+
+// BackendSummaryInfo returns a summary of the secret backends.
+// If we just want a model's in-use backends, it's the caller's
+// resposibility to provide the backendIDs in the filter.
+func (s *Service) BackendSummaryInfo(
+	ctx context.Context,
+	modelUUID coremodel.UUID, cloud cloud.Cloud, cred cloud.Credential,
+	reveal, all bool, names ...string,
```

Comment:
We want to avoid the practice of overloaded, general purpose methods. Methods should start with a verb.

So we could do:

GetAllBackendSummaryInfo()
GetNamedBackendSummaryInfo()

We only need cloud and cred for k8s right? And we have the model type check. Can we rethink the logic around this.

Reply:
All of these including the `Force` are existing code and copied from the apiserver package.
Happy to change.